src/secret_sharing/gf2_array.rs

```diff
@@ -0,0 +1,221 @@
+use std::{
+    fmt::Debug,
+    ops::{Add, AddAssign, Mul, Neg, Sub, SubAssign},
+};
+
+use bitvec::prelude::{BitSlice, Lsb0};
+use generic_array::GenericArray;
+use typenum::U1;
+
+use crate::{
+    ff::{Gf2, Serializable},
+    secret_sharing::{SharedValue, SharedValueArray},
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Gf2Array<const N: usize>([usize; N]);
```

Comment:
yea I agree, `const WORDS: usize` at least makes my brain stop and think before treating `N` as number of bits here.